src/ert/job_queue/driver.py

```diff
@@ -101,65 +106,130 @@ async def kill(self, realization: "RealizationState") -> None:
-        )
-        self._submit_processes[realization] = process
-
-        # Wait for submit process to finish:
-        output, error = await process.communicate()
-        print(output)  # FLAKY ALERT, we seem to get empty
-        print(error)
+        current_attempt = 0
+        while current_attempt < self._max_attempt:
+            current_attempt += 1
+            try:
+                process = await asyncio.create_subprocess_exec(
+                    *submit_cmd,
+                    stdout=asyncio.subprocess.PIPE,
+                    stderr=asyncio.subprocess.PIPE,
+                )
+                self._submit_processes[realization] = process
+
+                # Wait for submit process to finish:
+                output, error = await process.communicate()
+                if process.returncode != 0:
+                    logger.error(
+                        (
+                            f"bsub returned non-zero exitcode: {process.returncode}\n"
+                            f"Failed to get lsf job id from file: {output.decode()} \n {error.decode()}\n"
+                            f"bsub command                      : {submit_cmd}\n"
+                            "** ERROR ** Failed when submitting to LSF - will try again."
+                        )
+                    )
+                    time.sleep(self._SLEEP_PERIOD)
+                    continue
+
+                logger.info(output.decode())
+
+            except Exception as e:
+                logger.error(e)
+                time.sleep(self._SLEEP_PERIOD)
+                continue
 
-        try:
             lsf_id = str(output).split(" ")[1].replace("<", "").replace(">", "")
             self._realstate_to_lsfid[realization] = lsf_id
             self._lsfid_to_realstate[lsf_id] = realization
             realization.accept()
-            print(f"Submitted job {realization} and got LSF JOBID {lsf_id}")
-        except Exception:
-            # We should probably retry the submission, bsub stdout seems flaky.
-            print(f"ERROR: Could not parse lsf id from: {output!r}")
+            logger.info(f"Submitted job {realization} and got LSF JOBID {lsf_id}")
+            return
+
+        logger.error(
+            f"Tried submitting job {self._max_attempt} times, but it still failed"
+        )
+        raise RuntimeError("Maximum number of submit errors exceeded\n")
 
     async def poll_statuses(self) -> None:
         if self._currently_polling:
+            logger.debug("Already polling status elsewhere")
             return
+
         self._currently_polling = True
 
         if not self._realstate_to_lsfid:
             # Nothing has been submitted yet.
+            logger.warning("Skipped polling due to no jobs submitted")
             return
 
-        poll_cmd = ["bjobs"] + list(self._realstate_to_lsfid.values())
+        poll_cmd = [
+            self.get_option("BJOBS_CMD") if self.has_option("BJOBS_CMD") else "bjobs"
```

Comment:
I see from this line that we should have a `defaultdict` containing the driver defaults, but maybe that can be saved for a future pr

Reply:
Yes, agreed!